### PR TITLE
[TorchDISC] update pytorch api usages

### PR DESCRIPTION
**Pull request by fortianyou**
- add extra pytorch include header files
- Update to latest pytorch
- [TorchDISC] update pytorch api usages

### CODE REVIEW

File: torch_disc/torch_disc/csrc/disc_backend/backend_impl.cpp [hunk 1]
```diff
@@ -50,9 +49,7 @@ const std::set<int8_t> TSBackendDeviceType::supported_device_types_ = {
 class DISCBackendImpl : public torch::lazy::BackendImplInterface {
  public:
   DISCBackendImpl() : default_device_type_(at::kCPU) {
-    auto type = lazy_tensors::sys_util::GetEnvBool("LTC_TS_CUDA", false)
```
Comment (Yancey1989) [inline torch_disc/torch_disc/csrc/disc_backend/backend_impl.cpp:53]: why not using env to control device type?

File: torch_disc/torch_disc/csrc/disc_compiler/BUILD [hunk 2]
```diff
@@ -15,8 +15,8 @@ cc_library (
     "passes/register_disc_class.h",
     "passes/disc_class.h",
   ],
+  includes = ["../include"],
```
Comment (Yancey1989) [inline torch_disc/torch_disc/csrc/disc_compiler/BUILD:18]: maybe we should move `includes` dir to a meaningful dir, e.g. `torch_disc/external/pytorch`?

**Review by Yancey1989 — APPROVED** [review]
LGTM

**Review by fortianyou — COMMENTED** [review]
I will refactor it later since this is a temporary version.

**Pull request merged**